src/lightning/pytorch/trainer/setup.py

```diff
@@ -184,8 +193,17 @@ def _log_device_info(trainer: "pl.Trainer") -> None:
     if TPUAccelerator.is_available() and not isinstance(trainer.accelerator, TPUAccelerator):
         rank_zero_warn("TPU available but not used. You can set it by doing `Trainer(accelerator='tpu')`.")
 
-    if _IPU_AVAILABLE and not isinstance(trainer.accelerator, IPUAccelerator):
-        rank_zero_warn("IPU available but not used. You can set it by doing `Trainer(accelerator='ipu')`.")
+    if _IPU_AVAILABLE:
+        if not _LIGHTNING_GRAPHCORE_AVAILABLE:
+            raise ModuleNotFoundError(
+                "You are running on IPU machine but you have not installed `lightning-graphcore`"
+                f" extension is  {str(_LIGHTNING_GRAPHCORE_AVAILABLE)}."
+            )
```

Comment:
I think this shouldn't be an error. When user requests IPU but the package is not installed, this should already be an error in the accelerator connector. 

I would change the logic here to:
```py
if _IPU_AVAILABLE:
        if not _LIGHTNING_GRAPHCORE_AVAILABLE:
            rank_zero_warn("IPU available but not used. Run `pip install lightning-graphcore` and set `Trainer(accelerator='ipu')`)
        else:
            from lightning_graphcore import IPUAccelerator

            if not isinstance(trainer.accelerator, IPUAccelerator):
                rank_zero_warn("IPU available but not used. You can set it by doing `Trainer(accelerator='ipu')`.")
```

The same comment applies to HPU below, which needs to be fixed. Because otherwise, you can't run on the CPU.